specification/deviceregistry/resource-manager/Microsoft.DeviceRegistry/preview/2023-11-01-preview/asset.json

```diff
@@ -0,0 +1,663 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get an Asset": {
+            "$ref": "./examples/assets/Get_Asset.json"
+          },
+          "Get an Asset with Sync Status ": {
+            "$ref": "./examples/assets/Get_Asset_With_SyncStatus.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "assetNameParameter": {
+      "name": "assetName",
+      "description": "Asset name parameter.",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "pattern": "^[a-z0-9][a-z0-9-]*[a-z0-9]$",
+      "maxLength": 63,
+      "minLength": 3
+    }
+  },
+  "definitions": {
+    "AssetResponseList": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Asset"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      },
+      "description": "List of Assets."
+    },
+    "Asset": {
+      "type": "object",
+      "required": [
+        "extendedLocation"
+      ],
+      "properties": {
+        "properties": {
+          "type": "object",
+          "required": [
+            "assetEndpointProfileUri"
+          ],
+          "allOf": [
+            {
+              "$ref": "#/definitions/AssetProperties"
+            }
+          ],
+          "description": "Asset resource properties."
+        },
+        "extendedLocation": {
+          "type": "object",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "properties": {
+            "type": {
+              "type": "string",
+              "description": "The extended location type."
+            },
+            "name": {
+              "type": "string",
+              "description": "The extended location name."
+            }
+          },
+          "description": "The extended location."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Asset definition."
+    },
+    "AssetPatchPayload": {
+      "type": "object",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create",
+            "update"
+          ],
+          "description": "Resource tags."
+        },
+        "properties": {
+          "$ref": "#/definitions/AssetProperties",
+          "description": "Asset resource properties."
+        }
+      },
+      "description": "Asset definition for Patch operation."
+    },
+    "AssetProperties": {
+      "description": "Defines the asset properties.",
+      "type": "object",
+      "properties": {
+        "uuid": {
```

Comment:
From an ARM/Azure perspective, we typically use the ARM resource ID as the identifier for an ARM resource. I'm wondering if this is more of an internal concept that can possibly be hidden from users? I see another property is exposed regarding IDs of assets `externalAssetId`, consider minimizing what is exposed to customers - if possible.

Reply:
externalAssetId is exposed to the customer and is the addressable identifier of the asset on the edge. It is expected customers to provide this value. This follows the same pattern of devices where the externalDeviceId is used to address the external identity of the device.

Reply:
It makes sense for `externalAssetId` how about `uuid`?

Reply:
`uuid` is a global identifier for the Asset that aligns to the same approach Devices is using. We want an internal universal id that is not tied to the ARM resource ID to be part of the resource.